x/epochs/keeper/abci_test.go

```diff
@@ -4,211 +4,224 @@ import (
-						Duration:                time.Hour * 24 * 31,
-						CurrentEpoch:            0,
-						CurrentEpochStartHeight: suite.ctx.BlockHeight(),
-						CurrentEpochStartTime:   time.Time{},
-						EpochCountingStarted:    false,
-					},
-				},
-			})
+			// Check that custom genesis worked.
+			epochInfo, found = suite.network.App.EpochsKeeper.GetEpochInfo(ctx, "month")
+			suite.Require().True(found, "expected to find custom genesis data")
 
-			tc.fn()
+			tc.malleate(ctx)
 
-			suite.Require().Equal(epochInfo.Identifier, "monthly")
-			suite.Require().Equal(epochInfo.StartTime.UTC().String(), tc.expInitialEpochStartTime.UTC().String())
-			suite.Require().Equal(epochInfo.Duration, time.Hour*24*31)
-			suite.Require().Equal(epochInfo.CurrentEpoch, tc.expCurrentEpoch)
-			suite.Require().Equal(epochInfo.CurrentEpochStartHeight, tc.expCurrentEpochStartHeight)
-			suite.Require().Equal(epochInfo.CurrentEpochStartTime.UTC().String(), tc.expCurrentEpochStartTime.UTC().String())
-			suite.Require().Equal(epochInfo.EpochCountingStarted, true)
+			suite.Require().Equal(identifier, epochInfo.Identifier, "expected a different identifier")
+			suite.Require().Equal(month, epochInfo.Duration, "expected a different duration")
+			suite.Require().Equal(tc.expCurrentEpoch, epochInfo.CurrentEpoch, "expected a different current epoch")
+			suite.Require().Equal(tc.expCurrentEpochStartHeight, epochInfo.CurrentEpochStartHeight, "expected different current epoch start height")
+			suite.Require().Equal(tc.expCurrentEpochStartTime.UTC().String(), epochInfo.CurrentEpochStartTime.UTC().String(), "expected different current epoch start time")
+			suite.Require().Equal(tc.expCountingStarted, epochInfo.EpochCountingStarted, "expected different epoch counting started")
+			suite.Require().Equal(tc.expInitialEpochStartTime.UTC().String(), epochInfo.StartTime.UTC().String(), "expected a different initial start time")
 		})
 	}
 }
 
 func (suite *KeeperTestSuite) TestEpochStartingOneMonthAfterInitGenesis() {
-	// On init genesis, default epochs information is set
-	// To check init genesis again, should make it fresh status
-	epochInfos := suite.app.EpochsKeeper.AllEpochInfos(suite.ctx)
-	for _, epochInfo := range epochInfos {
-		suite.app.EpochsKeeper.DeleteEpochInfo(suite.ctx, epochInfo.Identifier)
-	}
-
 	now := time.Now()
 	week := time.Hour * 24 * 7
-	month := time.Hour * 24 * 30
-	initialBlockHeight := int64(1)
-	suite.ctx = suite.ctx.WithBlockHeight(initialBlockHeight).WithBlockTime(now)
-
-	epochs.InitGenesis(suite.ctx, suite.app.EpochsKeeper, types.GenesisState{
-		Epochs: []types.EpochInfo{
-			{
-				Identifier:              "monthly",
-				StartTime:               now.Add(month),
-				Duration:                time.Hour * 24 * 30,
-				CurrentEpoch:            0,
-				CurrentEpochStartHeight: suite.ctx.BlockHeight(),
-				CurrentEpochStartTime:   time.Time{},
-				EpochCountingStarted:    false,
-			},
+	month := time.Hour * 24 * 31
+
+	identifier := "month"
```

Comment:
loads of places where this is hardcoded 👀

Reply:
Yeah you're right, it come from old design. I created var for them but loacally but after you're review I moved everything to the setting file as const. Thanks!